com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -53,63 +53,26 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
-            throw e;
-        }
-        return StepPhase.POLL;
-    }
-
-    private void uploadApp(ExecutionWrapper execution, CloudApplicationExtended app, CloudFoundryOperations client) {
-        String processId = execution.getContext()
-            .getProcessInstanceId();
-        getStepLogger().trace("Started upload app step runnable for process \"{0}\"", processId);
-        try {
-            attemptToUploadApp(execution, app, client);
-        } catch (SLException | FileStorageException e) {
-            getStepLogger().error(e, Messages.ERROR_UPLOADING_APP, app.getName());
-            logException(execution.getContext(), e);
-            throw new SLException(e, e.getMessage());
-        } catch (CloudFoundryException cfe) {
-            CloudControllerException e = new CloudControllerException(cfe);
-            getStepLogger().error(e, Messages.ERROR_UPLOADING_APP, app.getName());
-            logException(execution.getContext(), e);
-            throw e;
-        } catch (Throwable e) {
-            Throwable eWithMessage = getWithProperMessage(e);
-            logException(execution.getContext(), eWithMessage);
-            if (e instanceof Exception) {
-                // only wrap Runtime & checked exceptions as Runtime ones
-                throw new RuntimeException(eWithMessage.getMessage(), eWithMessage);
-            } else {
-                // Errors and other should be handled elsewhere
-                throw e;
-            }
-        }
-        getStepLogger().trace("Upload app step runnable for process \"{0}\" finished", execution.getContext()
-            .getProcessInstanceId());
-    }
-
-    private void attemptToUploadApp(ExecutionWrapper execution, CloudApplicationExtended app, CloudFoundryOperations client)
-        throws FileStorageException {
-        AsyncExecutionState status = AsyncExecutionState.ERROR;
-        try {
             String appArchiveId = StepsUtil.getRequiredStringParameter(execution.getContext(), Constants.PARAM_APP_ARCHIVE_ID);
             String fileName = StepsUtil.getModuleFileName(execution.getContext(), app.getModuleName());
-            getStepLogger().debug("Uploading file \"{0}\" for application \"{1}\"", fileName, app.getName());
+            
+            getStepLogger().debug(Messages.UPLOADING_FILE_0_FOR_APP_1, fileName, app.getName());
             String uploadToken = asyncUploadFiles(execution, client, app, appArchiveId, fileName);
+            getStepLogger().debug(Messages.STARTED_ASYNC_UPLOAD_OF_APP_0, app.getName());
             execution.getContext()
                 .setVariable(Constants.VAR_UPLOAD_TOKEN, uploadToken);
-            getStepLogger().debug("Started async upload of application \"{0}\"", fileName, app.getName());
-            status = AsyncExecutionState.RUNNING;
-        } finally {
-            execution.getContext()
-                .setVariable(Constants.VAR_UPLOAD_STATE, status.name());
+        } catch (CloudFoundryException cfe) {

```

Comment:
Another question, before we used to catch Throwable and FileStorageException, for the FSE I suppose it is fine to catch just a SLException, but what about the Throwable, could there be such case?

Reply:
Any exception that is not handled in this class is handled in SyncActivitiStep. There's a catch Throwable there. :P

Reply:
verno be :d